File: 2015/python/02.py
def part1(p):
    s = 0
    for l,w,h in p:
        s += 2*l*w + 2*w*h + 2*h*l + l*w
    return s

def part2(p):
    s = 0
    for l,w,h in p:
        s += 2*l + 2*w + l*w*h
    return s

p = []

for l in open("../input/02").read().split('\n'):
    if l == '': break
    vls = [int(x) for x in l.split('x')]
    vls.sort()
    p.append((vls[0],vls[1],vls[2]))

print("Part 1:",part1(p))
print("Part 2:",part2(p))

File: 2015/cpp/02.cpp
// Advent of Code 2015, day 2
// (c) aichingert

#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <bits/stdc++.h>

std::vector<int> get(std::string line)
{
	std::vector<int> n;
	std::string cur;

	for (int i = 0; i < line.length(); ++i)
	{
		if (line[i] == '\n') break;
		if (line[i] == 'x') 
		{
			n.push_back(std::stoi(cur));
			cur = "";
		}
		else 
		{
			cur += line[i];
		}

	}
	
	if (line != "") n.push_back(std::stoi(cur));

	std::sort(n.begin(), n.end());
	return n;
}

int main(void)
{
	std::ifstream fp ("../input/02");
	std::string inp;
	int paper = 0;
	int ribbon = 0;

	if (fp.is_open())
	{
		while (fp)
		{
			std::getline(fp,inp);
			std::vector n = get(inp);

			if (n.size() > 0)
			{
				paper += 2*n[0]*n[1] + 2*n[1]*n[2] + 2*n[0]*n[2] + n[0]*n[1];
				ribbon += 2*n[0] + 2*n[1] + n[0]*n[1]*n[2];
			}
		}
	}

	std::cout << "Part 1: " << paper << "\n";
	std::cout << "Part 2: " << ribbon << "\n";

	return 0;
}
